src/Nri/Ui/SideNav/V2.elm

```diff
@@ -0,0 +1,450 @@
+
+@docs primary, secondary
+
+
+## Change the state
+
+@docs premiumDisplay
+
+-}
+
+import Accessibility.Styled exposing (..)
+import Accessibility.Styled.Style as Style
+import ClickableAttributes exposing (ClickableAttributes)
+import Css exposing (..)
+import Html.Styled
+import Html.Styled.Attributes as Attributes
+import Html.Styled.Events as Events
+import Nri.Ui
+import Nri.Ui.ClickableText.V3 as ClickableText
+import Nri.Ui.Colors.V1 as Colors
+import Nri.Ui.Data.PremiumDisplay as PremiumDisplay exposing (PremiumDisplay)
+import Nri.Ui.Fonts.V1 as Fonts
+import Nri.Ui.Html.Attributes.V2 as ExtraAttributes
+import Nri.Ui.Html.V3 exposing (viewJust)
+import Nri.Ui.Svg.V1 as Svg exposing (Svg)
+import Nri.Ui.UiIcon.V1 as UiIcon
+
+
+{-| Use `entry` to create a sidebar entry.
+-}
+type Entry route msg
+    = Entry (List (Entry route msg)) (EntryConfig route msg)
+    | Html (List (Html msg))
+
+
+{-| -}
+entry : String -> List (Attribute route msg) -> Entry route msg
+entry title attributes =
+    attributes
+        |> List.foldl (\(Attribute attribute) b -> attribute b) (build title)
+        |> Entry []
+
+
+{-| -}
+entryWithChildren : String -> List (Attribute route msg) -> List (Entry route msg) -> Entry route msg
+entryWithChildren title attributes children =
+    attributes
+        |> List.foldl (\(Attribute attribute) b -> attribute b) (build title)
+        |> Entry children
+
+
+{-| -}
+html : List (Html msg) -> Entry route msg
+html =
+    Html
+
+
+{-| -}
+type alias Config route msg =
+    { isCurrentRoute : route -> Bool
+    , routeToString : route -> String
+    , onSkipNav : msg
+    , css : List Style
+    }
+
+
+{-| -}
+view : Config route msg -> List (Entry route msg) -> Html msg
+view config entries =
+    styled nav
+        [ flexBasis (px 250)
+        , flexShrink (num 0)
+        , borderRadius (px 8)
+        , backgroundColor Colors.gray96
+        , padding (px 20)
+        , marginRight (px 20)
+        , batch config.css
+        ]
+        []
+        (viewSkipLink config.onSkipNav
+            :: List.map (viewSidebarEntry config []) entries
+        )
+
+
+viewSkipLink : msg -> Html msg
+viewSkipLink onSkip =
+    ClickableText.button "Skip to main content"
+        [ ClickableText.icon UiIcon.arrowPointingRight
+        , ClickableText.small
+        , ClickableText.css
+            [ Css.pseudoClass "not(:focus)"
+                [ Style.invisibleStyle
+                ]
+            ]
+        , ClickableText.onClick onSkip
+        ]
+
+
+viewSidebarEntry : Config route msg -> List Css.Style -> Entry route msg -> Html msg
+viewSidebarEntry config extraStyles entry_ =
+    case entry_ of
+        Entry children entryConfig ->
+            if entryConfig.premiumDisplay == PremiumDisplay.PremiumLocked then
+                viewLockedEntry extraStyles entryConfig
+
+            else if anyLinkDescendants (isCurrentRoute config) children then
+                div [ Attributes.css extraStyles ]
```

Comment:
Similar to the RadioButton changes, this is not introducing visual changes for PremiumUnlocked. If that is to land in another PR then we are good here.

Reply:
This visual change, on the other side, I think can be left for another PR, as it shouldn't change anything in the module's API